### PR TITLE
lib: couple block height and time

### LINKED ISSUE #35: Couple `block_height` and `block_time` together in DB and Bitcoin interfaces
In #29 we introduced the block time in addition to the block height in the `confirmed_coins` and `confirm_coins` result, as well as in the structures managing transaction (`GetTxRes` and `DbCoin` for instance).

Both fields are optional, but they should always either both be there or both not be. It would be cleaner to have a `BlockInfo` that itself is optional with `time` and `height` non-optional fields. So that we can avoid passing around confusing tuples, and avoid unwrapping on one field after we checked the other was present.

### PR DESCRIPTION
**Pull request by jp1ac4**
This PR aims to resolve #35.

I hope I'm on the right path here. Please let me know in case I need to make further changes.

Thanks.

### CODE REVIEW

**Comment by darosior** [comment]
Concept ACK. From a quick skim through the diff i think you are on the right path. Will review more in details later today.

File: src/database/sqlite/schema.rs [hunk 1]
```diff
@@ -153,13 +153,18 @@ pub struct DbSpendBlock {
     pub time: u32,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct DbBlockInfo {
+    pub height: i32,
+    pub time: u32,
+}
```
Comment (darosior) [inline src/database/sqlite/schema.rs:160]: Same, just rename `DbSpendBlock` into `DbBlockInfo`.

**Review by darosior — COMMENTED** [review]
Looks good to me, just a nit to reuse the existing "block info" data structures instead of introducing new ones.

File: src/database/mod.rs [hunk 2]
```diff
@@ -276,11 +276,25 @@ impl From<DbSpendBlock> for SpendBlock {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct BlockInfo {
+    pub height: i32,
+    pub time: u32,
+}
```
[the remark this answers is not in the document]
Comment (jp1ac4) [reply src/database/mod.rs:283]: Sure, I'll make this change.

**Comment by darosior** [comment]
Thanks!

utACK 89eb1572e955548886bf0c00c5e64b69f9f8cb32

**Comment by darosior** [comment]
For next time: it'd be cleaner for the two first commits to be merged, since the second one is undoing part of the diff from the first one.

**Comment by jp1ac4** [comment]
Thanks for your review, @darosior. Ah yes sorry, I was planning to squash at the end once you'd checked the changes. I should have mentioned that.

**Pull request merged**